sycl/doc/extensions/experimental/sycl_ext_intel_gpu_cache_config.asciidoc

```diff
@@ -0,0 +1,177 @@
+memory (SLM) and data cache. Developers may want more data cache or SLM based
+on their application. This extension adds runtime kernel property `gpu_cache_config`
+which provides a way to set the preferred cache configuration for a kernel.
+
+=== Feature Test Macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros".  Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_INTEL_GPU_CACHE_CONFIG` to one of the values defined
+in the table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's APIs the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version.  Base features are supported.
+|===
+
+=== Introduction
+
+This extension introduces new kernel property that can be applied to kernels
+using the mechanism defined in link:sycl_ext_oneapi_kernel_properties.asciidoc[sycl_ext_oneapi_kernel_properties].
+
+=== GPU Cache Config Property
+
+```c++
+namespace sycl::ext::intel::experimental {
+
+
+enum class gpu_cache_config_enum : std::uint16_t { large_slm, large_data };
+
+inline constexpr gpu_cache_config_enum large_slm =
+    gpu_cache_config_enum::large_slm;
+inline constexpr gpu_cache_config_enum large_data =
+    gpu_cache_config_enum::large_data;
+
+struct gpu_cache_config {
+  gpu_cache_config(gpu_cache_config_enum v) : value(v) {}
+  gpu_cache_config_enum value;
+};
+
+} // namespace sycl::ext::intel::experimental
+```
+
+The `gpu_cache_config` property provides a way to set the preferred cache
+configuration for a kernel. The following values are supported:
+
+ * `large_slm`: Prefer using more of a shared cache for
+    SLM and hence leaving less cache available for data.
+
+ * `large_data`: Prefer using less of a shared cache for SLM and
+    hence leaving more cache available for data.
+
+These property may be passed to any kernel invocation function (e.g.
+`parallel_for`) via the properties parameter.  At most, only one of these
+values may be passed to any single kernel invocation function.
+
+Backends that do not support this extension may accept and ignore this
+property.
+
+=== Adding a Property List to a Kernel Launch
+
+A simple example of using this extension  is shown below.
+
+The example assumes that the kernel will benefit from large SLM and hence uses the property
+`gpu_cache_config_large_slm`:
+
+```c++
+using namespace sycl::ext::intel::experimental;
+{
+  ...
+  properties kernel_properties{gpu_cache_config{large_slm}};
+
+  q.single_task(kernel_properties, [=] {
+    *a = *b + *c;
+  }).wait();
+}
+```
+
+=== Embedding Property into a Kernel
+
+The example below shows how the kernel from the previous section could be
+rewritten to leverage an embedded property list:
```

Comment:
Where is this "embedded property list" concept explained? I miss to see how `operator()` and `get(properties_tag)` are called.

Reply:
Thanks, I think it is useful to give a link to that document here.

Reply:
Provided the link.